### DISCUSSION TITLE
[enhancement] Add stoplist to the UI

### DISCUSSION THREAD

**Discussion by TuskarMA** [umputun/tg-spam#153] [Ideas]
Add the ability to edit the stoplist of words through the user interface. Moderators should have the ability to dynamically change it at any time.

**Comment by umputun** [comment]
I have never had such a need personally, as a blacklist for me is a very static thing not intended to be updated often. It is supposed to catch common phrases that spammers are using, and since I set one up a year ago for my group, I have not touched it since. Other filters, especially classifiers, are much better at handling the dynamic nature of spam.

Anyway, adding another tab with a simple editor should be easy to do.

**Comment by TuskarMA** [comment]
As practice over the last 4 years on other platforms (Discord and others) has shown, modern spammers are equipped with a whole arsenal of software that creates accounts, changes messages, and even checks them for similarity. Adding message comparison helps, but only for a short time, as spammers quickly find ways around it. That’s why I’d like to have the ability to dynamically add and change items in the blacklist to quickly respond to messages. I would happily add these PRs myself, but unfortunately, my knowledge of Go is almost zero. (I tried to clone the repo and modify something in the webapi locally, but after seeing the unfamiliar syntax and type system—an absolute nightmare for a Node.js developer like me—I quickly closed the IDE in fear😂)

In any case, you have a great project that, as I see it, has no real analogs (at least I haven’t found anything similar on GitHub). I think I’m not the only one who needs and will need the ability to edit the blacklist dynamically via the web UI in the future.

**Comment by umputun** [comment]
I get it, but what I'm saying is this: from a practical point of view and my experience fighting spam in two sizable TG groups, I had to update the blacklist only once, initially. As a tool to follow new spammers' tricks, adding spam samples to the classifier (which is already a feature in the UI) is a much more realistic and incomparably more effective way to deal with this. The intent for the blacklist was very simple - this is one of many heuristic-based methods intended to block very basic patterns, mostly set-and-forget kinds of things. I.e., if your chat has nothing to do with crypto, you can put this word on the blacklist. Or something like "send to personal account for more details," or whatever phrase they like to use. But the blacklist as a method to fight spam, especially dynamic and often updating, is not good. Classifier, similarity check, and AI checks do this job much, much better.

Adding the control to the UI for editing and an extra endpoint on the backend side is not hard; this is why I marked it as a "good first issue." Hopefully, someone will volunteer and make a PR for this.
